packages/common/src/hooks/useDownloadTrackButtons.ts

```diff
@@ -223,24 +222,17 @@ const makeDownloadOriginalButton = ({
     }
   }
 
-  if (track.download.cid) {
-    return {
-      ...config,
-      state: isLoggedIn
-        ? ButtonState.DOWNLOADABLE
-        : ButtonState.LOG_IN_REQUIRED,
-      onClick: () => {
-        if (!isLoggedIn) {
-          onNotLoggedInClick?.()
-        }
-        onDownload(track.track_id)
+  return {
+    ...config,
+    state: isLoggedIn ? ButtonState.DOWNLOADABLE : ButtonState.LOG_IN_REQUIRED,
+    onClick: () => {
```

Comment:
can totally ignore me here for this pr, but ive noticed this hook results in lots and lots of rerenders, wondering if the non callbacked onClick here is a potential culprit. i bet these changes will already help though

Reply:
ooo good eyes. lemme poke at it